oak-core/src/main/java/org/apache/jackrabbit/oak/query/ast/NodeLocalNameImpl.java

```diff
@@ -149,4 +149,13 @@ public OrderEntry getOrderEntry(SelectorImpl s, OrderingImpl o) {
             OrderEntry.Order.DESCENDING : OrderEntry.Order.ASCENDING);
     }
 
+    @Override
+    public String getOrderEntryPropertyName(SelectorImpl s) {
+        String fn = getFunction(s);
+        if (fn != null) {
+            return  QueryConstants.FUNCTION_RESTRICTION_PREFIX + fn;
+        }
```

Comment:
We have lots of repeated code, as this method is always the same. Shouldn't we rather write the method in the base class (if there is one)? To avoid repeating code.